collect_app/src/test/java/org/odk/collect/android/activities/viewmodels/MainMenuViewModelTest.java

```diff
@@ -0,0 +1,54 @@
+package org.odk.collect.android.activities.viewmodels;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+@RunWith(RobolectricTestRunner.class)
+public class MainMenuViewModelTest {
+
+    @Test
+    public void version_whenRelease_returnsSemanticVersionWithPrefix() {
+        MainMenuViewModel viewModel = new MainMenuViewModel(() -> "v1.1.7");
```

Comment:
Nice! These tests end up looking really nice.

I probably would have pushed all the methods now in the view model into the version provider class and kept the view model super thin but I do like how all this ends up looking.

Reply:
Yeah another step would be to extract a `VersionDescriptionParser` to deal with all this (and move the tests there) but I don't think that's worth it until the `ViewModel` is doing other things.

Reply:
Makes sense!